src/cpp/database/database.cpp

```diff
@@ -2779,6 +3133,1099 @@ DatabaseDump Database::dump_data_(
+                (*it).at(GUID_INFO_TAG),
+                participants_[participant_domain_id][participant_id],
+                topics_[topic_domain_id][topic_id]);
+
+            // Insert into database
+            EntityId entity_id = EntityId(string_to_int(it.key()));
+            insert_nts(entity, entity_id);
+
+            // Link endpoint with locator
+            for (auto it_loc = (*it).at(LOCATOR_CONTAINER_TAG).begin();
+                    it_loc != (*it).at(LOCATOR_CONTAINER_TAG).end();
+                    ++it_loc)
+            {
+                link_endpoint_with_locator_nts(entity->id, EntityId(string_to_int(*it_loc)));
+            }
+
+            // Load data and insert into database
+            load_data((*it).at(DATA_CONTAINER_TAG), entity);
+        }
+    }
+}
+
+void Database::load_data(
+        const DatabaseDump& dump,
+        const std::shared_ptr<DomainParticipant>& entity)
+{
+    // discovery_time
+    {
+        DatabaseDump container = dump.at(DATA_KIND_DISCOVERY_TIME_TAG);
+
+        // RemoteEntities iterator
+        for (auto remote_it = container.begin(); remote_it != container.end(); ++remote_it)
+        {
+            // Data iterator
+            for (auto it = container.at(remote_it.key()).begin(); it != container.at(remote_it.key()).end(); ++it)
+            {
+                DiscoveryTimeSample sample;
+
+                // std::chrono::system_clock::time_point
+                uint64_t src_ts = string_to_uint((*it).at(DATA_VALUE_SRC_TIME_TAG));
+                sample.src_ts = std::chrono::system_clock::time_point(std::chrono::steady_clock::duration(src_ts));
+
+                // std::chrono::system_clock::time_point
+                uint64_t time = string_to_uint(std::string((*it).at(DATA_VALUE_TIME_TAG)));
+                sample.time = std::chrono::system_clock::time_point(std::chrono::steady_clock::duration(time));
+
+                // EntityId
+                sample.remote_entity = EntityId(string_to_int(remote_it.key()));
+
+                string_to_int(std::string((*it).at(DATA_VALUE_REMOTE_ENTITY_TAG)));
+
+                // bool
+                sample.discovered = (*it).at(DATA_VALUE_DISCOVERED_TAG);
+
+                // Insert data into database
+                insert_nts(entity->domain->id, entity->id, sample, true);
+            }
+        }
+    }
+
+    // pdp_packets
+    {
+        DatabaseDump container = dump.at(DATA_KIND_PDP_PACKETS_TAG);
+
+        // Data iterator
+        for (auto it = container.begin(); it != container.end(); ++it)
+        {
+            PdpCountSample sample;
+
+            // std::chrono::system_clock::time_point
+            uint64_t time = string_to_uint((*it).at(DATA_VALUE_SRC_TIME_TAG));
+            sample.src_ts = std::chrono::system_clock::time_point(std::chrono::steady_clock::duration(time));
+
+            // uint64_t
+            sample.count = string_to_uint(to_string((*it).at(DATA_VALUE_COUNT_TAG)));
```

Comment:
I made it to throw exceptions when the .json has not the correct value. if the .json parses a value that must be a uint, it will convert it to int, avoiding the exception.